parametric/test_128_bit_traceids.py

```diff
@@ -0,0 +1,310 @@
+import pytest
+
+from parametric.utils.headers import make_single_request_and_get_inject_headers
+from parametric.utils.test_agent import get_span
+
+parametrize = pytest.mark.parametrize
+
+
+@pytest.mark.parametrize(
+    "library_env", [{"DD_TRACE_PROPAGATION_STYLE": "Datadog", "DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED": False,}],
+)
+def test_datadog_propagation_D001(test_agent, test_library):
+    """ Ensure that external 128-bit TraceIds are properly propagated in Datadog
+    headers.
+    """
+    with test_library:
+        headers = make_single_request_and_get_inject_headers(
+            test_library,
+            [
+                ["x-datadog-trace-id", "1234567890123456789"],
+                ["x-datadog-parent-id", "987654321"],
+                ["x-datadog-tags", "_dd.p.tid=640cfd8d00000000"],
+            ],
+        )
+
+    span = get_span(test_agent)
+    trace_id = span.get("trace_id")
+    dd_p_tid = span["meta"].get("_dd.p.tid")
+
+    assert trace_id == 1234567890123456789
+    validate_dd_p_tid(dd_p_tid)
+    assert dd_p_tid == "640cfd8d00000000"
+
+
+@pytest.mark.parametrize(
+    "library_env", [{"DD_TRACE_PROPAGATION_STYLE": "Datadog", "DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED": False,}],
+)
+def test_datadog_generation_64_bit_D002(test_agent, test_library):
+    """Ensure that 64-bit TraceIds are properly generated, propagated in
+    datadog headers, and populated in trace data.
+    """
+    with test_library:
+        headers = make_single_request_and_get_inject_headers(test_library, [])
+
+    header_trace_id = headers["x-datadog-trace-id"]
+    span = get_span(test_agent)
+    span_trace_id = span.get("trace_id")
+    dd_p_tid = span["meta"].get("_dd.p.tid")
+
+    assert len(header_trace_id) < 21
+    assert int(header_trace_id, 10) == span_trace_id
+    assert dd_p_tid is None
+
+
+@pytest.mark.skip_library("dotnet", "not implemented")
+@pytest.mark.skip_library("golang", "not implemented")
+@pytest.mark.skip_library("java", "not implemented")
+@pytest.mark.skip_library("nodejs", "not implemented")
+@pytest.mark.skip_library("python", "not implemented")
+@pytest.mark.skip_library("python_http", "not implemented")
+@pytest.mark.parametrize(
+    "library_env", [{"DD_TRACE_PROPAGATION_STYLE": "Datadog", "DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED": True,}],
+)
+def test_datadog_generation_128_bit_D003(test_agent, test_library):
```

Comment:
non-blocking nit: I was thinking about a slight rename of the tests that makes the intent a little clearer to me. What do you think?
- `test_<format>_128_bit_propagation_<TEST#>`
- `test_<format>_128_bit_generation_enabled_<TEST#>`
- `test_<format>_128_bit_generation_disabled_<TEST#>`